controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1443,18 +1439,17 @@ bool master_thread::handle_intel_slave_join(
           (slave_version_s.build_number > master_version_s.build_number)))) {
         LOG(INFO) << "slave_version > master_version, sending "
                      "ACTION_CONTROL_VERSION_MISMATCH_NOTIFICATION";
-        auto response = message_com::add_intel_vs_data<
-            beerocks_message::cACTION_CONTROL_VERSION_MISMATCH_NOTIFICATION>(cmdu_tx, tlv_vs);
-        if (response == nullptr) {
+        auto version_mismatch_vs_tlv = message_com::add_vs_tlv<
```

Comment:
Here, you're changing the behaviour: previously there was a VS TLV containing just a VERSION_MISMATCH_NOTIFICATION. With this change, you have a VS TLV containing both a JOIN_RESPONSE and a secont VS TLV with VERSION_MISMATCH_NOTIFICATION.

I think the creation of the join_response has to move down into the branches that actually use it.

Reply:
The previous behavior is wrong too. What is the point of adding new beerocks header with mismatch data like that? The version mismatch is part of the join response. I don't know why it is implemented that way. On the Intel server, it is part of the join response and not a separate tlv.
For now I'll revert it to how it was but I need to check with @tomereli about this.

Reply:
Probably code that had never been executed...